opal/util/output.c

```diff
@@ -63,6 +63,8 @@ static opal_output_stream_t verbose;
 static char *output_dir = NULL;
 static char *output_prefix = NULL;
 
+static opal_output_pmix_cleanup_fn_t output_pmix_cleanup_fn = NULL;
```

Comment:
unlike `pmix`, `output` is now part of the core functionality. why are we using pmix in the name of this type in the `core` library ?

Reply:
That is correct. `output` is part of the OPAL core, but `pmix` is **not** part of the OPAL core. The `output.c` code had this code in it:
https://github.com/open-mpi/ompi/blob/b2c26c5993c895a8aab8b7d582aea9bcca9a89b8/opal/util/output.c#L786-L787

The call was inserted to tell the host RM not to clean up this file when cleaning up session directories and whatnot when the job completes. The `open_file` function may be called whenever a new framework/component opens a new output handle.

So I needed to find a way to get a pointer to the [`opal_pmix_register_cleanup()`](https://github.com/open-mpi/ompi/blob/main/opal/mca/pmix/base/pmix_base_fns.c#L460-L523) OPAL wrapper around the `PMIx_Job_control_nb ` PMIx function into the OPAL core without linking to it (so tools wouldn't complain about the missing symbol). 

To do this I added a function registration mechanism. I added the `pmix` name as a reminder of how it connects.